### PR TITLE
Pre-disabled Row

### PR DESCRIPTION
**Pull request by harshzalavadiya**
Revised PR of https://github.com/jbetancur/react-data-table-component/pull/188
Since there was so many merge conflicts so I think it's better to have clean PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,6 +132,7 @@ When the breakpoint is reached the column will be hidden. These are the built-in
 | onRowSelected | func | no |  | callback to access the row selected state ({ allSelected, selectedCount, selectedRows }). <br /> **note** It's highly recommended that you memoize the callback that you pass to `onRowSelected` if it updates the state of your parent component. This prevents `DataTable` from unnescessary re-renders everytime your parent component is re-rendered |
 | selectableRowsComponent | func | no |  | Override the default checkbox component - must be passed as a function (e.g. `Checkbox` not `<Checkbox />`) |
 | selectableRowsComponentProps | object | no |  | Additional props you want to pass to `selectableRowsComponent`. See [Advanced Selectable Component Options](#advanced-selectable-component-options) to learn how you can override indeterminate state |
+| selectableRowsDisabledField | string | no |  | Disable some rows from selection based on `bool` field |
```
Comment (jbetancur) [inline README.md]: I may reword this but I can do this on another PR since I am fixing some things in the readme

File: src/DataTable/TableColCheckbox.js [hunk 2]
```diff
@@ -14,9 +14,12 @@ const TableColStyle = styled(CellBase)`
 `;
 
 const TableCol = () => {
-  const { dispatch, data, selectedRows, allSelected, selectableRowsComponent, selectableRowsComponentProps } = useTableContext();
+  const { dispatch, data, selectedRows, allSelected, selectableRowsComponent, selectableRowsComponentProps, selectableRowsDisabledField } = useTableContext();
   const indeterminate = selectedRows.length > 0 && !allSelected;
-  const handleSelectAll = () => dispatch({ type: 'SELECT_ALL', rows: data });
+  const rows = data.filter(
+    datam => !datam[selectableRowsDisabledField],
```
Comment (jbetancur) [inline src/DataTable/TableColCheckbox.js]: datum

File: src/DataTable/TableCellCheckbox.js [hunk 3]
```diff
@@ -13,9 +13,10 @@ const TableCellCheckboxStyle = styled(CellBase)`
 `;
 
 const TableCellCheckbox = ({ name, row }) => {
-  const { dispatch, data, selectedRows, selectableRowsComponent, selectableRowsComponentProps } = useTableContext();
+  const { dispatch, data, selectedRows, selectableRowsComponent, selectableRowsComponentProps, selectableRowsDisabledField } = useTableContext();
   const handleOnRowSelected = () => dispatch({ type: 'ROW_SELECTED', row, rows: data });
   const isRowSelected = useMemo(() => selectedRows.some(srow => srow === row), [row, selectedRows]);
+  const isRowDisabled = selectableRowsDisabledField ? row[selectableRowsDisabledField] : false;
```
Comment (jbetancur) [inline src/DataTable/TableCellCheckbox.js]: I think you can shorten this to:
```
 const isRowDisabled = row[selectableRowsDisabledField];
```

**Comment by jbetancur** [comment]
NICE! Just 2 minor things and I will get this merged today. 

As for the tests I will take care of adding them

**Comment by jbetancur** [comment]
Thanks for the contribution @harshzalavadiya!

There is a bit of an edge case here, such as what if all rows are disabled? Then the select all should be disabled as well. I'll address this on my PR for the test cases.

Once that is done I'll do a new feature release 3.1.0

**Comment by jbetancur** [comment]
released 3.1.0!
